utils/validation.js

```diff
@@ -45,51 +46,48 @@ const validateToken = async (req, res, uid) => {
         return res.status(401).json(getResponseJSON('Account already exists', 401));
     }
 
-    if(req.query.token && req.query.token.trim() !== "") {
-        const token = req.query.token.trim();
+    const pin = req.query.pin?.trim();
+    const token = req.query.token?.trim();
+    if (!pin && !token) {
+        return res.status(400).json(getResponseJSON('Bad request: token/pin required', 400));
+    }
 
-        const { verifyToken } = require('./firestore');
-        const isValid = await verifyToken(token);
+    if (token) {
+      const {
+        verifyToken,
+        linkParticipanttoFirebaseUID,
+      } = require('./firestore');
+      const { isDuplicateAccount, isValid: isValidToken, docId } = await verifyToken(token);
```

Comment:
merging `verifyPIN()` and `verifyToken()` makes sense, but why keep the two functions just to setup to parameter we pass to `verifyTokenOrPin()`? Seems like an unnecessary middle man.